### ISSUE TITLE
DateTime.ParseExact handles "yyy" incorrectly.

### ISSUE THREAD

**Issue by isonil** [dotnet/runtime#51860]
DateTime.ParseExact handles "yyy" incorrectly.

### Description

DateTime.ParseExact requires the year to be 3-digits when using "yyy" despite "yyy" normally accepting 3-4 digits as per the [documentation](https://docs.microsoft.com/en-US/dotnet/standard/base-types/custom-date-and-time-format-strings).

Other 1-2 digit parts, like "H" for 1-2 digits hours, or "m" for 1-2 digits minutes work correctly.

Repro case:
```
using System;
using System.Globalization;
					
public class Program
{
	public static void Main()
	{
		DateTime.ParseExact("2021:04:26 10:19:57", "yyyy:M:d H:m:s", CultureInfo.InvariantCulture);
		
		Console.WriteLine("OK");
		
		DateTime.ParseExact("2021:04:26 10:19:57", "yyy:M:d H:m:s", CultureInfo.InvariantCulture);
	}
}
```
Output:
```
OK
Run-time exception (line 12): String was not recognized as a valid DateTime.
```
Try it [here](https://dotnetfiddle.net/mKqx6U)

### Configuration

Tested on .NET 4.7.2, .NET 5, and .NET Core 3.1

**Comment by SimonCropp** [comment]
my vote would be to change the documentation

**Comment by tarekgh** [comment]
I agree to fix the documentation as changing that can break other cases. https://github.com/dotnet/runtime/pull/115507#issuecomment-2877227989
